classes/manager.php

```diff
@@ -132,9 +132,9 @@ public static function get_total_weight() {
      * @return bool
      * @throws \dml_exception
      */
-    public static function is_factorid_valid($factorname, $factorid, $user) {
+    public static function is_factorid_valid($factorid, $user) {
         global $DB;
-        $recordowner = $DB->get_field('factor_'.$factorname, 'userid', array('id' => $factorid));
+        $recordowner = $DB->get_field('tool_mfa', 'userid', array('id' => $factorid));
 
         if (!empty($recordowner) && $recordowner == $user->id) {
```

Comment:
Nit: I've move the `$recordowner == $user->id` into the sql where clause and then simplify this if to just a return